### PR TITLE
Remove sizes attribute when responsive images are disabled

### LINKED ISSUE #1389: Accurate sizes improvement didn't account for the disable filter for sizes
## Bug Description

WordPress core has a mechanism to disable responsive images through some filters.

```php
add_filter( 'wp_calculate_image_srcset', '__return_empty_array' );
add_filter( 'wp_calculate_image_sizes', '__return_false' );
```

If we disable responsive images, the Enhance Responsive Image plugin still adds the sizes and srcset values. We should fix this.

## Steps to reproduce

1. Add `add_filter( 'wp_calculate_image_sizes', '__return_false' )` in your functions.php 
2. Add image through Image block
3. See image markup.

## Screenshots

Screenshot 2024-07-23 at 5 53 15 PM

### PR DESCRIPTION
**Pull request by mukeshpanchal27**
## Summary

Fixes #1389

Check the issue description for testing details.

### CODE REVIEW

**Comment by mukeshpanchal27** [comment]
As WordPress core does not add sizes attribute for thumbnail size, In https://github.com/WordPress/performance/pull/1399/commits/e83ac2d898300ea3b987a75db48b62c6b5423561 i have to remove it from tests.

Is there any specific reason for it 🤔

**Comment by adamsilverstein** [comment]
> As WordPress core does not add sizes attribute for thumbnail size, In [e83ac2d](https://github.com/WordPress/performance/commit/e83ac2d898300ea3b987a75db48b62c6b5423561) i have to remove it from tests.

were tests failing without this change?

> 
> Is there any specific reason for it 🤔

I was wondering as well. Maybe for HDPI versions of thumbnails (x2) if that were a thing? For picture element we want srcset for fallbacks, but not sure why we would want sizes.

**Review by adamsilverstein — APPROVED** [review]
Nice!

File: plugins/auto-sizes/hooks.php [hunk 1]
```diff
@@ -156,6 +156,15 @@ function auto_sizes_improve_image_sizes_attributes( string $content ): string {
 		return $content;
 	}
 
+	// Bail early if the responsive images are disabled.
+	if ( null === $processor->get_attribute( 'sizes' ) ) {
+		$processor->remove_attribute( 'data-needs-sizes-update' );
+		$processor->remove_attribute( 'data-align' );
+		$processor->remove_attribute( 'data-resize-width' );
```
Comment (westonruter) [inline plugins/auto-sizes/hooks.php]: Minor point, but there is some code duplication here. You could refactor this a bit to not have to maintain two sets of attributes to remove:

```suggestion
	$remove_data_attributes = static function () use ( $processor ): void {
		$processor->remove_attribute( 'data-needs-sizes-update' );
		$processor->remove_attribute( 'data-align' );
		$processor->remove_attribute( 'data-resize-width' );
	};

	// Bail early if the responsive images are disabled.
	if ( null === $processor->get_attribute( 'sizes' ) ) {
		$remove_data_attributes();
```

And then `$remove_data_attributes` could be reused below:

https://github.com/WordPress/performance/blob/e83ac2d898300ea3b987a75db48b62c6b5423561/plugins/auto-sizes/hooks.php#L218-L223

**Comment by mukeshpanchal27** [comment]
@westonruter @adamsilverstein or @swissspidy Could you please review and merge it. Thanks!

**Review by westonruter — APPROVED** [review]

**Pull request merged**